### PR TITLE
[IMS-2 ]Fixed registration

### PR DESCRIPTION
**Pull request by fredthuo1**
Fixed the changes suggested in the last code review

### CODE REVIEW

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 1]
```diff
@@ -82,9 +85,9 @@ public GenericResponse resetPassword(final HttpServletRequest request, @RequestP
         return new GenericResponse(messages.getMessage("message.resetPasswordEmail", null, request.getLocale()));
     }
 
-    // Save Password
+    // Reset password after forgetting old password
     @PostMapping("/user/savePassword")
-    public GenericResponse savePassword(final Locale locale, @Valid PasswordDto passwordDto) {
+    public GenericResponse savePassword(@RequestBody final Locale locale, @RequestBody @Valid PasswordDto passwordDto) {
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:90]: change to recovery

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 2]
```diff
@@ -52,25 +54,26 @@ public RegistrationRestController() {
 
     // Registration
     @PostMapping("/user/registration")
-    public GenericResponse registerUserAccount(@Valid final UserDto accountDto, final HttpServletRequest request) {
+    public GenericResponse registerUserAccount(@RequestBody @Valid final UserDto accountDto, final HttpServletRequest request) {
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:57]: registration/verification

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 3]
```diff
@@ -52,25 +54,26 @@ public RegistrationRestController() {
...
+    @GetMapping("/user/sendRegistrationToken")
+    public GenericResponse sendRegistrationToken(final HttpServletRequest request, @RequestParam("token") final String existingToken) {
         final VerificationToken newToken = userService.generateNewVerificationToken(existingToken);
         final User user = userService.getUser(newToken.getToken());
         mailSender.send(constructResendVerificationTokenEmail((getAppUrl(request).toString()), request.getLocale(), newToken, user));
         return new GenericResponse(messages.getMessage("message.resendToken", null, request.getLocale()));
     }
 
-    // Reset password
+    // Send link to email after forgetting password to reset password. The link will redirect user to /user/savePassword
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:76]: account/registration/oneTimePassword

**Comment by allanjoroge** [comment]
All good!

**Pull request closed without merging**